### PR TITLE
Add (optional) support for listening for wemo events

### PR DESCRIPTION
**Pull request by tomwilkie**
Does not add any dependancies.

### CODE REVIEW

**Comment by balloob** [comment]
I don't have anything to test this with. As far as looking at the code without running it, it looks good. I'll go ahead and merge it. I've also added you as a collaborator in case you have more ideas.

I see your awesomenation project and it looks like we're building something [very similar](https://github.com/balloob/home-assistant) :-)

**Comment by tomwilkie** [comment]
Thanks!  Do you want me to write a little unit test?  It works with the switch (is pretty cool - turning the switch on form the ios app and the status update in awesomation!)  I'll be making it work with the motion sensor this weekend.

Yes I've been following your project :-)  Its look very good; I started Awesomation as I wanted to pursue a slightly different architecture, with different goals - I wanted this system to be completely hands off and learning - ie I wasn't interested in having a UI at all.  Have you seen my little blog?  I've explained some of the things I've been working on in more detail there: http://homeawesomation.blogspot.co.uk/

Other than working on things like pywemo together, do you think there is anything else we could collaborate on?

**Comment by balloob** [comment]
Unit test would be a nice to have, not a must have.

I haven't read your blog yet but I'll make sure to check it out.

With Home Assistant I want to offer home control (the UI), allowing automation rules to be setup (home automation) and the smart home (hands off system). I have all the pieces into place (history, reproduce states) except for the logic what to do when.

For collaboration: we can share the Python libraries to talk to devices (like pywemo) but we can also collaborate on [device discovery](https://github.com/balloob/netdisco).

**Comment by tomwilkie** [comment]
Will do a unit test this weekend.

I've mainly been focusing on the logic of what to do when - currently its
limited to turning lights on when someone is in a room, turning the heating
on when someone is home, and (almost) having the music and TV following you
around the house.  Next I want to look at security-ish behaviours (no one
home but movement detected?  learning your pattern and turning some lights
on when not home?)

Discovery looks interesting - I've got a ton of threads sitting there doing
basically the same thing.  Will check it out and see how I can use it.

Cheers

Tom
